src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs

```diff
@@ -206,6 +206,15 @@ private static void SwapViews(_View oldView, _View newView)
 			{
 				parentAsContentControl.Content = newView;
 			}
+			else if (newView is Page newPage && oldView is Page oldPage)
+			{
+				// In the case of Page, swapping the actual page is not supported, so we
+				// need to swap the content of the page instead. This can happen if the Frame
+				// is using a native presenter which does not use the `Frame.Content` property.
+				oldPage.DataContext = null;
```

Comment:
Approving, but I'm not sure about this specific line.

Reply:
I'm not sure either. We may need to adjust it to propagate the inherited DataContext from the Frame instead, I'll try check for that part and see if it's more appropriate.